### ISSUE TITLE
Road network from OSM?

### ISSUE THREAD

**Issue by crisjf** [CityScope/CS_GAMABrix#4]
In the Road Traffic model, the example uses a street network imported from a shapefile [here](https://gama-platform.github.io/wiki/RoadTrafficModel_step3). Is there a way of importing the road network for the given table from GAMA?

**Comment by agrignard** [comment]
Yes there is this feature in GAMA and it's actually already written in this repo https://github.com/CityScope/CS_Simulation_GAMA. It would be great to have this feature. I would say not by defautl as the first toy model should be only dealing with what is present in the GEOGRID  but if we have the possibility to link it to an OSM layer that would be great.

The code is already written here https://github.com/CityScope/CS_Simulation_GAMA/blob/master/CS_CityScope_GAMA/models/VisualizationToolKit/OSM_Loader_Model.gaml

That's also the kind of reason why I was wondering if we should create a new repo for GAMABrix as there are some utilities that could be reuse in the initial repo by just including it.

**Comment by crisjf** [comment]
It seems that this model doesn't download the OSM data from the OSM API, but rather loads it from a file. Is this right @agrignard ?

**Comment by agrignard** [comment]
No i think you have the 2 option, either you have an osm and if  not it gives the possibility to donwlaod the osm by providing the top left and top right of the table 

`````
osm_file osmfile;
		if (file_exists(osm_file_path)) {
			osmfile  <- osm_file(osm_file_path, filtering);
		} else {
			point top_left <- CRS_transform({0,0}, "EPSG:4326").location;
			point bottom_right <- CRS_transform({shape.width, shape.height}, "EPSG:4326").location;
			string adress <-"http://overpass.openstreetmap.ru/cgi/xapi_meta?*[bbox="+top_left.x+"," + bottom_right.y + ","+ bottom_right.x + "," + top_left.y+"]";
			osmfile <- osm_file<geometry> (adress, filtering);
		}
````
